### PR TITLE
Add account activation, password reset, edit&destroy user actions

### PR DESCRIPTION
**Pull request by andreydro**

### CODE REVIEW

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user,         only: [:edit, :update]
+  before_action :valid_user,       only: [:edit, :update]
```
Comment (kavalenka) [inline app/controllers/password_resets_controller.rb]: Too many spaces before only

File: app/controllers/account_activations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class AccountActivationsController < ApplicationController
+  def edit
```
Comment (kavalenka) [inline app/controllers/account_activations_controller.rb]: I think method name should be ``activate``

File: app/helpers/account_activations_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (kavalenka) [inline app/helpers/account_activations_helper.rb]: Empty file

File: app/helpers/password_resets_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module PasswordResetsHelper
```
Comment (kavalenka) [inline app/helpers/password_resets_helper.rb]: Empty file

File: app/models/user.rb [hunk 5]
```diff
@@ -8,11 +8,12 @@ class User < ApplicationRecord
   validates :email, presence: true, length: { maximum: 255 },
             format: { with: VALID_EMAIL_REGEX },
             uniqueness: { case_sensitive: false }
-  validates :password, presence: true, length: { minimum: 6 }
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
```
Comment (kavalenka) [inline app/models/user.rb:11]: How does it work together: presence: true and allow_nil: true?

File: app/views/layouts/_error_messages.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
```
Comment (kavalenka) [inline app/views/layouts/_error_messages.html.erb]: I think this partial should not only be connected with @user model.

File: app/views/users/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<div class="container">
+  <div class="row">
+    <div class="col-md-3">
+      <div class="settings-nav">
```
Comment (kavalenka) [inline app/views/users/edit.html.erb]: Good part of code for nav partial.

File: spec/controllers/account_activations_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (kavalenka) [inline spec/controllers/account_activations_controller_spec.rb]: Empty file

File: spec/helpers/account_activations_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (kavalenka) [inline spec/helpers/account_activations_helper_spec.rb]: Empty file

**Review by kavalenka — CHANGES_REQUESTED** [review]

File: app/controllers/account_activations_controller.rb (hunk 2 above)
Comment (andreydro) [reply app/controllers/account_activations_controller.rb]: I already have ```activate``` method in user model

File: app/models/user.rb (hunk 5 above)
Comment (andreydro) [reply app/models/user.rb:11]: ```presence: true``` doesn't give a chance to sign up without password
```allow_nil: true``` give a possibility to update user info (name, email) without password

File: app/controllers/account_activations_controller.rb (hunk 2 above)
Comment (kavalenka) [reply app/controllers/account_activations_controller.rb]: So, what's the problem to have the same here? It would be much more clearly to understand.

**Review by kavalenka — APPROVED** [review]

**Pull request merged**